examples/demo4.py

```diff
@@ -133,16 +150,29 @@ def main(argv):
         # Take the (e1, e2) shape parameters from the catalog as well.
         psf = psf.shear(e1=cat.getFloat(k,2), e2=cat.getFloat(k,3))
 
-        # Galaxy is a bulge + disk with parameters taken from the catalog:
-        disk = galsim.Exponential(flux=0.6, half_light_radius=cat.getFloat(k,5))
-        disk = disk.shear(e1=cat.getFloat(k,6), e2=cat.getFloat(k,7))
+        # Galaxy is a bulge + disk(+knots) with parameters taken from the catalog:
+
+        # put some fraction of the disk light into knots of star formation
 
-        bulge = galsim.DeVaucouleurs(flux=0.4, half_light_radius=cat.getFloat(k,8))
+        disk_hlr = cat.getFloat(k,5)
+        disk_e1 = cat.getFloat(k,6)
+        disk_e2 = cat.getFloat(k,7)
+        smooth_disk = galsim.Exponential(flux=smooth_disk_frac, half_light_radius=disk_hlr)
+
+        knots = galsim.RandomWalk(n_knots, disk_hlr, flux=knot_frac)
+
+        disk = galsim.Add([smooth_disk, knots])
+        disk = disk.shear(e1=disk_e1, e2=disk_e2)
+
+        # the rest of the light goes into the bulge
+        bulge_hlr = cat.getFloat(k,8)
+        bulge = galsim.DeVaucouleurs(flux=0.4, half_light_radius=bulge_hlr)
```

Comment:
for consistency with the rest of the updates, maybe call this ```bulge_frac``` instead of 0.4?  That way people can change some of the numbers up top and everything down here gets updated appropriately.